### PR TITLE
Add scaled float to the scripting fields API

### PR DESCRIPTION
**Pull request by jdconrad**
This adds the mapped type scaled float the scripting fields API using double as the base type.

### CODE REVIEW

File: modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/50_script_doc_values.yml [hunk 1]
```diff
@@ -1340,6 +1341,45 @@ setup:
...
+          rest_total_hits_as_int: true
+          body:
+            sort: [ { rank: asc } ]
+            script_fields:
+              field:
+                script:
+                  source: "field('scaled_float').get(1, 0.0)"
+    - match: { hits.hits.0.fields.field.0: 0.0 }
+    - match: { hits.hits.1.fields.field.0: 0.0 }
+    - match: { hits.hits.2.fields.field.0: 2.5 }
```
Comment (stu-elastic) [inline modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/50_script_doc_values.yml:1381]: Shouldn't we get back -350 and 250 here?

**Review by stu-elastic — APPROVED** [review]
Can you add unit tests to `FloatDocValuesFieldTests`?  Looking good otherwise.

File: modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/50_script_doc_values.yml (hunk 1 above)
Comment (jdconrad) [reply modules/lang-painless/src/yamlRestTest/resources/rest-api-spec/test/painless/50_script_doc_values.yml:1381]: The scaling factor is for how the field is indexed behind the scenes. The values here are returned as expected.

**Comment by jdconrad** [comment]
@stu-elastic Thanks for the review. I added unit tests as requested. Will commit as soon as CI passes.

**Pull request merged**